File: app/src/release/java/djordjeh/architecture/mvp/ui/task/TaskModule.java
package djordjeh.architecture.mvp.ui.task;

import dagger.Binds;
import dagger.Module;
import djordjeh.architecture.mvp.dagger.scope.FragmentChildScope;

@Module
public abstract class TaskModule {

    @Binds
    @FragmentChildScope
    abstract ContractTask.View bindView(TaskFragment fragment);

    @Binds
    @FragmentChildScope
    abstract ContractTask.Presenter bindPresenter(TaskPresenter presenter);
}

File: app/src/main/java/djordjeh/architecture/mvp/ui/task/TaskModule.kt
package djordjeh.architecture.mvp.ui.task

import dagger.Binds
import dagger.Module
import djordjeh.architecture.mvp.dagger.scope.FragmentChildScope

@Module
abstract class TaskModule {

    @Binds
    @FragmentChildScope
    internal abstract fun bindView(fragment: TaskFragment): ContractTask.View

    @Binds
    @FragmentChildScope
    internal abstract fun bindPresenter(presenter: TaskPresenter): ContractTask.Presenter
}
